MetaMask/metamask-mobile
# fix: Verify third party details causes to watch the asset undesirably

## **Description**
When approving a token and you click on `Verify third party details` it adds the token to the wallet. This PR prevents calling `addToken` on the token approve screen.

## **Manual testing steps**
1. Go to the test dapp
2. Deploy an ERC721/ERC20 token
3. Click Approve
4. Click Verify Third party details
5. Cancel
6. Go to the Home wallet
7. You shouldn't see a new token added to your wallet.

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e072c203-f63d-4115-bf86-768408fe083e

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/54408225/00e92cf0-948f-47d4-bb7a-b604d361279d

### **After**


https://github.com/MetaMask/metamask-mobile/assets/29962968/cba721d4-8b69-49f9-a1ae-ed22f1fa7ad4


## **Related issues**

_Fixes #7178 

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
